core/pkg/service/flag-evaluation/json_codec_test.go

```diff
@@ -0,0 +1,139 @@
+//nolint:wrapcheck
+package service
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func TestJSONCodec_Marshaling(t *testing.T) {
+	originalMap := map[string]interface{}{
+		"val1": false,
+		"val2": 0.0,
+		"val3": "",
+		"val4": true,
+		"val5": 1.0,
+		"val6": "hi!",
+	}
+	tests := map[string]struct {
+		message    func() (any, error)
+		marshaller func(any) ([]byte, error)
+		wantErr    bool
+		wantMap    map[string]interface{}
+	}{
+		"Marshal no error": {
+			message: func() (any, error) {
+				return structpb.NewStruct(originalMap)
+			},
+			marshaller: func(message any) ([]byte, error) {
+				jsonCodec := jsonCodec{}
+				return jsonCodec.Marshal(message)
+			},
+			wantErr: false,
+			wantMap: originalMap,
+		},
+		"MarshalStable no error": {
+			message: func() (any, error) {
+				return structpb.NewStruct(originalMap)
+			},
+			marshaller: func(message any) ([]byte, error) {
+				jsonCodec := jsonCodec{}
+				return jsonCodec.MarshalStable(message)
+			},
+			wantErr: false,
+			wantMap: originalMap,
+		},
+		"MarshalAppend no error": {
+			message: func() (any, error) {
+				return structpb.NewStruct(originalMap)
+			},
+			marshaller: func(message any) ([]byte, error) {
+				jsonCodec := jsonCodec{}
+				return jsonCodec.MarshalAppend(nil, message)
+			},
+			wantErr: false,
+			wantMap: originalMap,
+		},
+		"Marshal not valid message": {
+			message: func() (any, error) {
+				return map[string]interface{}{}, nil
+			},
+			marshaller: func(message any) ([]byte, error) {
+				jsonCodec := jsonCodec{}
+				return jsonCodec.Marshal(nil)
+			},
+			wantErr: true,
+			wantMap: map[string]interface{}{},
+		},
+		"MarshalStable not valid message": {
+			message: func() (any, error) {
+				return map[string]interface{}{}, nil
+			},
+			marshaller: func(message any) ([]byte, error) {
+				jsonCodec := jsonCodec{}
+				return jsonCodec.MarshalStable(nil)
+			},
+			wantErr: true,
+			wantMap: map[string]interface{}{},
+		},
+		"MarshalAppend not valid message": {
+			message: func() (any, error) {
+				return map[string]interface{}{}, nil
+			},
+			marshaller: func(message any) ([]byte, error) {
+				jsonCodec := jsonCodec{}
+				return jsonCodec.MarshalAppend(nil, nil)
+			},
+			wantErr: true,
+			wantMap: map[string]interface{}{},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			jsonMap := map[string]interface{}{}
+			message, err := tt.message()
+			if err != nil {
+				t.Errorf("Got error creating message: %v", err.Error())
+			}
+			bytes, err := tt.marshaller(message)
+			if tt.wantErr && err == nil {
+				t.Errorf("Expected error but not none")
```

Comment:
optional suggestion: To keep consistent with the other assertions, you could also use `require.NotNil(t, err)`, or `require.NotNilf(t, err, "Expected error but not none")`

Reply:
ty!

Reply:
(I meant to type "got none" also)